sycl/include/CL/sycl/accessor.hpp

```diff
@@ -834,6 +834,9 @@ class accessor :
 #endif // __SYCL_DEVICE_ONLY__
 
     size_t Result = 0;
+#ifdef __SYCL_DEVICE_ONLY__
+#pragma unroll
+#endif // __SYCL_DEVICE_ONLY__
```

Comment:
Why is it needed for device code only?

Reply:
We guarantee, that we compile device code with clang, where the pragma is known. Host compiler can any C++ compiler and not all of them know the pragma.

Reply:
Unknown pragmas are supposed to be ignored (with a warning, though 'system' headers don't warn!), so I don't know if the preprocessing is necessary.  

Additionally, limiting us to device only seems like too big hammer here, I would expect if you're limiting this being there,  you would do it more to detect the right compiler.  Specifically, I know GCC and ICC also support this.

Reply:
I've added compiler checks. If it won't work, I guess I'll add unrolling via template magic.